wdl-lint/src/rules/redundant_input_assignment.rs

```diff
@@ -0,0 +1,111 @@
+//! A lint rule for flagging redundant input assignments
+
+use std::fmt::Debug;
+
+use rowan::ast::AstNode;
+use wdl_ast::AstNodeExt;
+use wdl_ast::AstToken;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::SyntaxElement;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+use wdl_ast::v1::CallStatement;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the Redundant Input Assignment rule.
+const ID: &str = "RedundantInputAssignment";
+
+/// Create a "Redundant Input Assignment" diagnostic.
+fn redundant_input_assignment(span: Span, name: &str) -> Diagnostic {
+    Diagnostic::note("redundant input assignment")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix(format!("can be shortened to `{}`", name))
+}
+
+/// Detects a redundant input assignment.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct RedundantInputAssignment(Option<SupportedVersion>);
+
+impl Rule for RedundantInputAssignment {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags redundant input assignments."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Input assignments that are redundant can be shortened. For example, `{ input: a = a }` \
+         can be shortened to `{ input: a }`."
```

Comment:
```suggestion
        "Redundant input assignments can be shortened in WDL versions >=v1.1. For example, `{ input: a = a }` can be shortened to `{ input: a }`."
```
This will need to be run through `cargo fmt` again after committing the wording change 👍